test/checks/shared/heading-whitespace-only.js

```diff
@@ -0,0 +1,214 @@
+describe('heading-whitespace-only', function() {
```

Comment:
You're not testing combinations of things, multiple chars that should fail, chars that should fail together with chars that pass, chars mixed with break elements. Most of these tests can be integration tests if you move the chars and breaking space elements into an option object.